### PR TITLE
Handle git-lfs separately

### PR DESCRIPTION
**Pull request by 6543**
initial: #24

bug: https://codeberg.org/Codeberg-CI/feedback/issues/83

### CODE REVIEW

File: plugin_test.go [hunk 1]
```diff
@@ -97,6 +99,27 @@ var commits = []struct {
 		file: "README",
 		data: "Hello World!\n\nsomething is changed!\n",
 	},
+	// chekout with lfs
```
Comment (MFTabriz) [inline plugin_test.go:102]: this should read "checkout"

File: plugin_test.go [hunk 2]
```diff
@@ -97,6 +99,27 @@ var commits = []struct {
...
+		path:     "test-assets/woodpecker-git-test-lfs",
+		clone:    "https://github.com/test-assets/woodpecker-git-test-lfs.git",
+		event:    "push",
+		commit:   "69d4dadb4c2899efb73c0095bb58a6454d133cef",
+		ref:      "refs/heads/main",
+		file:     "4M.bin",
+		dataSize: 4194304,
+		lfs:      true,
+	},
+	// chekout with lfs skip
```
Comment (MFTabriz) [inline plugin_test.go:113]: this should read "checkout"

**Comment by 6543** [comment]
ok now it works as intended: 
- NO lfs: https://ci.codeberg.org/6543/lfs-ci-test/build/22/3
- lfs: https://ci.codeberg.org/6543/lfs-ci-test/build/23/3

test image = woodpeckerci/plugin-git:test


but tests still fail :/

**Comment by 6543** [comment]
test `TestCloneNonEmpty` did first fetch and checkout **with** lfs and next without, but in that case it already does exist ... so it fail

now we first check aganist non lfs and enable it in next run first.

File: plugin_test.go [hunk 3]
```diff
@@ -97,6 +98,27 @@ var commits = []struct {
 		file: "README",
 		data: "Hello World!\n\nsomething is changed!\n",
 	},
+	// ### test lfs, please do not change order, otherwise TestCloneNonEmpty will fail ###
+	// chekout with lfs skip
```
Comment (MFTabriz) [inline plugin_test.go:102]: this should read "checkout"

Please make sure you fixed it before marking the issue as resolved!

Comment (6543) [inline plugin_test.go:102]:
```suggestion
	// checkout with lfs skip
```

File: plugin.go [hunk 4]
```diff
@@ -23,7 +23,10 @@ type Plugin struct {
 
 const customCertTmpPath = "/tmp/customCert.pem"
 
-var defaultEnvVars = []string{"GIT_TERMINAL_PROMPT=0"}
+var defaultEnvVars = []string{
+	"GIT_TERMINAL_PROMPT=0", // dont wait for user imput
```
Comment (lafriks) [inline plugin.go:27]: imput -> input

File: plugin_test.go [hunk 5]
```diff
@@ -97,6 +98,27 @@ var commits = []struct {
...
+	{
+		path:     "test-assets/woodpecker-git-test-lfs",
+		clone:    "https://github.com/test-assets/woodpecker-git-test-lfs.git",
+		event:    "push",
+		commit:   "69d4dadb4c2899efb73c0095bb58a6454d133cef",
+		ref:      "refs/heads/main",
+		file:     "4M.bin",
+		dataSize: 132,
+	},
+	// checkout with lfs
```
Comment (MFTabriz) [inline plugin_test.go:112]: LFS does not seem to be tested here!

File: plugin.go (hunk 4 above)
Comment (6543) [inline plugin.go:27]:
```suggestion
	"GIT_TERMINAL_PROMPT=0", // don't wait for user input
```

File: plugin_test.go (hunk 5 above)
Comment (6543) [reply plugin_test.go:112]: hmm ba7d69980776138258b09c17ee5f909fd4ff3827 did missed it

**Review by MFTabriz — APPROVED** [review]
lgtm! thanks for taking care of this.

**Review by lafriks — APPROVED** [review]

**Pull request merged**